cranelift/codegen/src/isa/aarch64/lower.isle

```diff
@@ -695,3 +695,222 @@
       (value_reg (alu_rs_imm_logic (ALUOp.EorNot64) $I64 x y)))
 
 (rule (lower (has_type $I128 (bxor_not x y))) (i128_alu_bitop (ALUOp.EorNot64) x y))
+
+;;;; Rules for `ishl` ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+;; Shift for i8/i16/i32.
+(rule (lower (has_type (fits_in_32 ty) (ishl x y)))
+      (value_reg (do_shift (ALUOp.Lsl32) ty (put_in_reg x) y)))
+
+;; Shift for i64.
+(rule (lower (has_type $I64 (ishl x y)))
+      (value_reg (do_shift (ALUOp.Lsl64) $I64 (put_in_reg x) y)))
+
+;; Shift for i128.
+(rule (lower (has_type $I128 (ishl x y)))
+      (lower_shl128 (put_in_regs x) (value_regs_get (put_in_regs y) 0)))
+
+;;     lsl     tmp3, src_lo, amt
+;;     lsl     tmp2, src_hi, amt
+;;     mvn     inv_amt, amt
+;;     lsr     tmp1, src_lo, #1
+;;     lsr     tmp1, tmp1, inv_amt
+;;     tst     amt, #0x40
+;;     orr     tmp2, tmp2, tmp1
+;;     csel    dst_hi, tmp3, tmp2, ne
+;;     csel    dst_lo, xzr, tmp3, ne
```

Comment:
This sequence (copied from the original non-SSA code) reuses temps, so it's a bit confusing in context here... could you rewrite it in terms of the variable binding names in the `let` below?